Analytics/Analytics/Base/Configuration.swift

```diff
@@ -20,8 +20,17 @@ public class Configuration {
     var gzipEnabled: Bool
     var storage: Storage
     var flushPolicies: [FlushPolicy]
+    var collectDeviceId: Bool
     
-    public init(writeKey: String, dataPlaneUrl: String, controlPlaneUrl: String = Constants.defaultControlPlaneUrl, logger: Logger = SwiftLogger(logLevel: Constants.defaultLogLevel), optOut: Bool = false, gzipEnaabled: Bool = Constants.defaultGZipStatus, storage: Storage? = nil, flushPolicies: [FlushPolicy] = Constants.defaultFlushPolicies) {
+    public init(writeKey: String,
+                dataPlaneUrl: String,
+                controlPlaneUrl: String = Constants.defaultControlPlaneUrl,
+                logger: Logger = SwiftLogger(logLevel: Constants.defaultLogLevel),
+                optOut: Bool = false,
+                gzipEnaabled: Bool = Constants.defaultGZipStatus,
+                storage: Storage? = nil,
+                flushPolicies: [FlushPolicy] = Constants.defaultFlushPolicies,
+                collectDeviceId: Bool = true) {
```

Comment:
Let's not hard code this value. Define it in the Constants file and use it from there.

```suggestion
                collectDeviceId: Bool = Constant.collectDeviceId) {
```